### PR TITLE
Prevent deleting last passwordless device with no password

### PR DESCRIPTION
**Pull request by Joerger**
Prevent users from deleting their last passwordless device if they have no other means of logging in, even if passwordless is currently disabled. This way, if passwordless is re-enabled, the user will be able to log in.

I found this small logic edge case while creating https://github.com/gravitational/teleport/pull/47426 and figured it may as well be covered.

Honestly, after giving this more thought and seeing the change on its merit, this may be a distinction without a difference. In practice the user will probably need to be reset anyways, except for the hard to imagine edge case:
1) admin temporarily disabled passwordless login
2) the user has an active login session
3) the user deletes their passkey in, due to confusion or otherwise
4) admin re-enables passworldess, user is now locked out, which could have been avoided

Changelog: Update logic around deleting last passkey to prevent edge case lockouts.

### CODE REVIEW

**Review by smallinsky — APPROVED** [review]

File: lib/auth/auth.go
Comment (codingllama) [inline lib/auth/auth.go:3883]: As you said, it's a pretty weird edge case, but I think I'm OK with the if essentially saying "is last passkey && has no password".

**Review by codingllama — APPROVED** [review]
Thanks for the small, focused PR.

**Comment by codingllama** [comment]
Note: depends on #47426. Please wait for the base PR to be merged before merging this one (naturally).

- #47426

**Pull request merged**
